# Благотворительный фонд Рауль

## Техническое задание проекта
Необходимот построить дэшборд. Дэшборд располагается на ресурсе, доступном сотрудникам организации БФ Рауль (https://raoul.spb.ru/).
Дэшборд отображает динамику и структуру трудоустройств.
Параметры динамики: помесячно, поквартально, ежегодно.
Параметры структуры: принадлежность благополучателя к одному из сегментов, ответственная НКО, регион, пол благополучателя, возраст, тип трудоустройства (временное, постоянное с сопровождением, постоянное без сопровождения), тип работодателя (внутренняя защищённая площадка, внешняя защищённая площадка, открытый рынок).
Параметры трудоустройства: от 1 дня, от 1 месяца

## Описание данных

Данные были обезличены для соблюдения 152-ФЗ о защите данных.
- **id_кандидата** - идентификационный номер благополучателя
- **пол**
- **возраст**
- **дата_добавления** - дата добавления благополучателя в картотеку
- **инвалидность** - одно из оснований получения помощи от фонда
- **сиротский_опыт** - одно из оснований получения помощи от фонда
- **адаптивная_образовательная_программа** - одно из оснований получения помощи от фонда
- **начало_трудоустройства**
- **завершение_трудоустройства**
- **1+_месяц_работы**
- **тип_площадки**
- **тип_трудоустройства**
- **тип_опыта_работы**
- **регион_код** - зашифрованный регион оказания услуги
- **провайдер_код** - зашифрованный контрагент фонда

## Знакомство с данными

In [ ]:
# импорт библиотек
import pandas as  pd
import requests
from urllib.parse import urlencode

import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
# Чтение файла "БФРауль_Данные_по_трудоустроенным.xls" с Яндекс диска

# используем api
base_url = 'https://cloud-api.yandex.net/v1/disk/public/resources/download?'
public_key = 'https://disk.yandex.ru/i/KeSNZWlrtWCHFQ'

# получаем url
final_url = base_url + urlencode(dict(public_key=public_key))
response = requests.get(final_url)
download_url = response.json()['href']

# загружаем файл в df
download_response = requests.get(download_url)
df = pd.read_excel(download_url)

In [ ]:
display(df.info())
display(df.head())
display(df.describe())
print('Количество пропусков:\n{}'.format(df.isna().sum()))
print()
print('Количество дубликатов: {}'.format(df.duplicated().sum()))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 259 entries, 0 to 258
Data columns (total 16 columns):
 #   Column                                    Non-Null Count  Dtype         
---  ------                                    --------------  -----         
 0   Unnamed: 0                                259 non-null    int64         
 1   id_кандидата                              259 non-null    int64         
 2   пол                                       259 non-null    object        
 3   возраст                                   259 non-null    int64         
 4   дата_добавления                           258 non-null    datetime64[ns]
 5   инвалидность                              259 non-null    object        
 6   сиротский_опыт                            259 non-null    object        
 7   адаптированная_образовательная_программа  259 non-null    object        
 8   начало_трудоустройства                    259 non-null    datetime64[ns]
 9   завершение_трудоустройства      

None

Unnamed: 0  id_кандидата      пол  возраст дата_добавления инвалидность  \
0           0             1  Женский       25      2018-12-06       ИСТИНА   
1           1             2  Мужской       28      2021-09-22       ИСТИНА   
2           2             3  Женский       31      2016-07-08       ИСТИНА   
3           3             4  Мужской       34      2022-10-07       ИСТИНА   
4           4             4  Мужской       34      2022-10-07       ИСТИНА   

  сиротский_опыт адаптированная_образовательная_программа  \
0           ЛОЖЬ                                     ЛОЖЬ   
1           ЛОЖЬ                                     ЛОЖЬ   
2           ЛОЖЬ                                     ЛОЖЬ   
3           ЛОЖЬ                                     ЛОЖЬ   
4           ЛОЖЬ                                     ЛОЖЬ   

  начало_трудоустройства завершение_трудоустройства 1+_месяц_работы  \
0             2023-01-16                        NaT          ИСТИНА   
1             2023-01-09                        NaT          ИСТИНА   
2             2023-04-18                 2023-06-30          ИСТИНА   
3             2022-11-21                 2022-12-30          ИСТИНА   
4             2023-03-22                 2023-03-30            ЛОЖЬ   

  тип_площадки               тип_трудоустройства тип_опыта_работы регион_код  \
0      Внешняя  Открытый рынок без сопровождения        Повторный     reg200   
1      Внешняя   Открытый рынок с сопровождением        Повторный     reg200   
2      Внешняя  Открытый рынок без сопровождения        Повторный     reg200   
3      Внешняя  Открытый рынок без сопровождения        Повторный     reg200   
4      Внешняя  Открытый рынок без сопровождения        Повторный     reg200   

  провайдер_код  
0      prov3500  
1      prov3500  
2      prov3500  
3      prov3500  
4      prov3500

Unnamed: 0  id_кандидата     возраст
count  259.000000    259.000000  259.000000
mean   129.000000    122.293436   22.359073
std     74.911058     70.736428    5.574585
min      0.000000      1.000000   14.000000
25%     64.500000     61.500000   18.000000
50%    129.000000    124.000000   22.000000
75%    193.500000    182.500000   25.000000
max    258.000000    242.000000   39.000000

Количество пропусков:
Unnamed: 0                                    0
id_кандидата                                  0
пол                                           0
возраст                                       0
дата_добавления                               1
инвалидность                                  0
сиротский_опыт                                0
адаптированная_образовательная_программа      0
начало_трудоустройства                        0
завершение_трудоустройства                  174
1+_месяц_работы                               0
тип_площадки                                  1
тип_трудоустройства                           1
тип_опыта_работы                              1
регион_код                                    0
провайдер_код                                 0
dtype: int64

Количество дубликатов: 0


В поле "завершение_трудоустройства" имеется 174 пропуска. Заказчик объяснил это тем, что люди продолжают работать в текущее время.
Рассмотрим единичные пропуски в четырех полях:

In [ ]:
df[(df.дата_добавления.isna()==True) | (df.тип_площадки.isna()==True) | (df.тип_трудоустройства.isna()==True) | (df.тип_опыта_работы.isna()==True) ]


Unnamed: 0  id_кандидата      пол  возраст дата_добавления инвалидность  \
209         209           197  Женский       19      2021-05-01         ЛОЖЬ   
226         226           213  Мужской       17             NaT         ЛОЖЬ   

    сиротский_опыт адаптированная_образовательная_программа  \
209         ИСТИНА                                     ЛОЖЬ   
226         ИСТИНА                                     ЛОЖЬ   

    начало_трудоустройства завершение_трудоустройства 1+_месяц_работы  \
209             2023-08-01                        NaT          ИСТИНА   
226             2023-08-01                        NaT          ИСТИНА   

    тип_площадки        тип_трудоустройства тип_опыта_работы регион_код  \
209          NaN                        NaN              NaN     reg204   
226      Внешняя  Временное трудоустройство           Первый     reg204   

    провайдер_код  
209      prov3504  
226      prov3504

Заказчик объяснил эти пропуски ошибками при вводе данных. Оставляем без изменений.

По поводу полей:
- инвалидность
- сиротский_опыт
- адаптированная_образовательная_программа

есть пожелание к заказчику преобразовать их в логический тип (True/False) для удобства.

Поле "1+_месяц_работы" в принципе неактуально, поскольку в дашборде этот показатель вычисляется каждые новые сутки на основе текущей даты и даты начала трудоустройства.


Поскольку заказчик предоставил лишь небольшой срез данных для настройки работы дашборда, рассчитывая в дальнейшем самостоятельно дополнять датасет, не вижу смысла подробно анализировать их. Поэтому переходим непосредственно к дашборду:

https://datalens.yandex.ru/wd3aj0eigvhal-otchet-po-trudoustroystvu-bf-raul
